samples/gui/lvgl/src/main.c

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2018 dXplore
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <device.h>
+#include <display.h>
+#include <lvgl.h>
+#include <stdio.h>
+#include <string.h>
+#include <zephyr.h>
+
+#define SYS_LOG_DOMAIN "main"
+#define SYS_LOG_LEVEL SYS_LOG_LEVEL_DEBUG
+#include <logging/sys_log.h>
+
+#define SCREEN_WIDTH 320
+#define SCREEN_HEIGHT 240
+
+static void disp_flush(int32_t x1, int32_t y1, int32_t x2, int32_t y2,
+		       const lv_color_t *color_p);
+
+static void lvgl_tick_handler(struct k_timer *timer_id);
+
+static struct device *display_dev;
+
+static s64_t time_stamp;
+
+K_TIMER_DEFINE(lvgl_tick, lvgl_tick_handler, NULL);
+
+void main(void)
+{
+	u32_t count = 0;
+	char count_str[11] = {0};
+	lv_disp_drv_t disp_drv;
+	lv_obj_t *hello_world_label;
+	lv_obj_t *count_label;
+
+	display_dev = device_get_binding("ILI9340");
+
+	if (display_dev == NULL) {
+		SYS_LOG_ERR("device not found.  Aborting test.");
+		return;
+	}
+
+	lv_init();
+
+	time_stamp = k_uptime_get_32();
+	k_timer_start(&lvgl_tick, K_MSEC(1), K_MSEC(1));
+
+	lv_disp_drv_init(&disp_drv);
+	disp_drv.disp_flush = disp_flush;
+	lv_disp_drv_register(&disp_drv);
+
+	hello_world_label = lv_label_create(lv_scr_act(), NULL);
+	lv_label_set_text(hello_world_label, "Hello world!");
+	lv_obj_align(hello_world_label, NULL, LV_ALIGN_CENTER, 0, 0);
+
+	count_label = lv_label_create(lv_scr_act(), NULL);
+	lv_obj_align(count_label, NULL, LV_ALIGN_IN_BOTTOM_MID, 0, 0);
+
+	display_on(display_dev);
+
+	while (1) {
+		if ((count % 100) == 0) {
+			sprintf(count_str, "%d", count/100);
+			lv_label_set_text(count_label, count_str);
+		}
+		lv_task_handler();
+		k_sleep(10);
+		++count;
+	}
+}
+
+void disp_flush(int32_t x1, int32_t y1, int32_t x2, int32_t y2,
+		const lv_color_t *color_p)
+{
+	u8_t *data = (u8_t *)color_p;
+	u16_t w = x2 - x1 + 1;
+	u16_t h = y2 - y1 + 1;
+	u32_t pixels = w * h;
+	u32_t to;
+	u32_t from;
+	u32_t idx;
+
+	for (idx = 0, to = 3, from = 4; idx < pixels;
+	     ++idx, to += 3, from += 4) {
+		data[to] = data[from];
+		data[to + 1] = data[from + 1];
+		data[to + 2] = data[from + 2];
+	}
+	display_write_bitmap(display_dev, x1, y1, w, h,
+			PIXEL_FORMAT_RGB_888, (void *) data);
+
+	lv_flush_ready();
+}
+
+void lvgl_tick_handler(struct k_timer *timer_id)
```

Comment:
Moved from #9149 
> @pizi-nordic
> Is this function generic? If so, could we move it to the lvgl integration layer?

Reply:
Yes it is and it only used to keep track of the system time in lvgl. So I'm wondering if it is not better to reimplement the lvgl functions ([lv_hal_tick.c](https://github.com/littlevgl/lvgl/blob/fc05fad2770d62de8085c88e1a2b15c313576909/lv_hal/lv_hal_tick.c)) to get the system time and time delta by a zephyr specific implementation.

Reply:
lvgl v5.2-rc provides option to use zephyr system time, code has been update to make use of this feature